Reduce unsafeness in PDF and assorted code

annevk opened the pull request:
#### e50830ecb937696505f2eb4ecd4e8976fb62836f
<pre>
Reduce unsafeness in PDF and assorted code
<a href="https://bugs.webkit.org/show_bug.cgi?id=294960">https://bugs.webkit.org/show_bug.cgi?id=294960</a>

Reviewed by Charlie Wolfe.

Apply <a href="https://github.com/WebKit/WebKit/wiki/Safer-CPP-Guidelines">https://github.com/WebKit/WebKit/wiki/Safer-CPP-Guidelines</a>

Canonical link: <a href="https://commits.webkit.org/296716@main">https://commits.webkit.org/296716@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/259ed3d64f507dada11f3c424dd7ef45af602582

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/109400 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/29058 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/19487 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/114605 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/59638 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/111363 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/29737 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/37646 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/83146 "Passed tests") | [⏳ 🧪 win-tests](https://ews-build.webkit.org/#/builders/Win-Tests-EWS "Waiting to run tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/112348 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/23684 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/98538 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/63603 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/23069 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/16681 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/59226 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/93053 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/16723 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/117717 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/36441 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/26975 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/92155 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/36812 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/94800 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/91970 "Passed tests") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/36907 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/14650 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/32250 "Built successfully") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/17651 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/36334 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/41810 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/36005 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/39344 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/37709 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

cdumez commented on Source/WebKit/WebProcess/Plugins/PDF/UnifiedPDF/PDFDataDetectorOverlayController.mm:
```diff
@@ -71,7 +71,7 @@
 }
 
-PageOverlay& PDFDataDetectorOverlayController::installOverlayIfNeeded()
+Ref<PageOverlay> PDFDataDetectorOverlayController::installProtectedOverlayIfNeeded()
```
Which is this returning a `Ref<>`? We are not transferring ownership. This doesn't not look like a step in the right direction.

annevk replied:
This change is to address https://build.webkit.org/results/Apple-Sequoia-Safer-CPP-Checks/296659%40main%20%284013%29/scan-build-output/StaticAnalyzer/WebKit/StaticAnalyzerReports/report-PDFDataDetectorOverlayController.mm-handleMouseEvent-44-d83ac6.html#EndPath

I could also make the protected variant wrap the non-protected variant, but I thought that if there's only one caller that might be a bit redundant?

cdumez replied:
Oh, you added `protected` in the name! I missed that during review. This is fine then.

cdumez commented on Source/WebKit/WebProcess/WebPage/mac/PageBannerMac.mm:
```diff
@@ -79,30 +79,31 @@
     // We can hide the banner by removing the parent layer that hosts it.
     if (m_type == Header)
-        m_webPage->corePage()->setHeaderHeight(0);
+        protectedWebPage()->corePage()->setHeaderHeight(0);
```
I don't understand this change. `corePage()` is trivial so we should not need to protect m_webPage here.

annevk replied:
Sorry, I should protect `corePage()` instead here and below.

cdumez commented on Source/WebKit/WebProcess/WebPage/mac/PageBannerMac.mm:
```diff
@@ -79,30 +79,31 @@
+        protectedWebPage()->corePage()->setHeaderHeight(0);
     else if (m_type == Footer)
-        m_webPage->corePage()->setFooterHeight(0);
+        protectedWebPage()->corePage()->setFooterHeight(0);
```
I don't understand this change. `corePage()` is trivial so we should not need to protect m_webPage here.

annevk committed:
Reduce unsafeness in PDF and assorted code
https://bugs.webkit.org/show_bug.cgi?id=294960

Reviewed by Charlie Wolfe.

Apply https://github.com/WebKit/WebKit/wiki/Safer-CPP-Guidelines

Canonical link: https://commits.webkit.org/296716@main
